### ISSUE TITLE
expect().to.have.length(N) does not work

### ISSUE THREAD

**Issue by bajtos** [prodatakey/dirty-chai#5]
The following code does not work for me:

```js
expect([1], 'msg').to.have.length(1);
// Uncaught TypeError: number is not a function
```

Workaround:

```js
expect([1], 'msg').length(1);
```

Versions: io.js 1.4.2, chai 2.1.1, dirty-chai 1.1.0.

**Comment by joshperry** [comment]
Hrrrm, this is difficult because chai does not distinguish between asserting properties and simple chaining properties. I have not found a good way to differentiate so I hook all the chain properties. This causes them to return a chaining function which causes this problem (`length` is an immutable property on `function`).

Let me think about this and see if I can come up with a good fix, if not I can just add an exclusion list.

**Comment by joshperry** [comment]
This is fixed in 1.2.0

Don't love the solution but it works at least for the built-in chai properties like `to` and `have`.

**Comment by bajtos** [comment]
Lovely, I'll give it a try when writing new tests. Thank you for a quick fix, it's much appreciated!
